File: 104. Maximum Depth of Binary Tree.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func max(x, y int) int {
	if x > y {
		return x
	}
	return y
}

func depthFirstSearch(node *TreeNode, depth int) int {
	if node == nil {
		return depth
	}
	depth++
	return max(depthFirstSearch(node.Left, depth), depthFirstSearch(node.Right, depth))

}

func maxDepth(root *TreeNode) int {
	return depthFirstSearch(root, 0)
}

func main() {

}

File: 104. Maximum Depth of Binary Tree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root == None: return 0
        else: return max(1+self.maxDepth(root.left),1+self.maxDepth(root.right))
